### PR TITLE
#162685657  Document API using swagger

### PR DESCRIPTION
**Pull request by VivianDoreen**
**What does this PR do?**
This PR finishes a feature "document API using swagger"

**Description of Task to be completed?**
Document API using swagger

**What are the relevant Pivotal Tracker stories?**
[#162685657](https://www.pivotaltracker.com/n/projects/2232496)

**How should it be manually tested?**
Run each of the following commands in your terminal in the respective order.

1. git clone https://github.com/andela/Ah-backend-guardians.git
2. view the cloned repo in VS code
3. In the terminal of VSCode, git checkout ft-document-api-162685657
4. Activate the virtual environment. (.venv/bin/activate)
5. Pip install -U drf-yasg
6. Run the app python manage.py runserver
7. Hit this URL to access the swagger endpoint. http://127.0.0.1:8000/api/swagger/
8. Login in or register to get the token
9. Click on the authorize button at the top and type "Bearer token"
10. Start testing the other endpoints

You should be able to get back the output as in the screenshot.

Screenshot
image

After clicking on the Authorize button
screen shot 2019-01-17 at 12 19 12
 pm

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ env:
 install:
     - pip install -r requirements.txt
     - pip install coveralls
+    - pip install -U drf-yasg
```
Comment (archbishop12) [inline .travis.yml]: @VivianDoreen Please remove this from here.

File: authors/apps/authentication/migrations/0002_remove_user_is_staff.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 2.1.5 on 2019-01-16 16:06
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (archbishop12) [inline authors/apps/authentication/migrations/0002_remove_user_is_staff.py]: @VivianDoreen Remove these migration files from your pr. You are not making any model changes.

File: authors/apps/authentication/migrations/0003_user_is_staff.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 2.1.5 on 2019-01-16 16:06
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (archbishop12) [inline authors/apps/authentication/migrations/0003_user_is_staff.py]: @VivianDoreen Same here

File: requirements.txt [hunk 4]
```diff
@@ -13,16 +16,25 @@ django-cors-middleware==1.3.1
 django-extensions==2.1.4
 django-heroku==0.3.1
 django-nose==1.4.6
+django-rest-swagger==2.2.0
```
Comment (archbishop12) [inline requirements.txt]: @VivianDoreen You have this here but you are not using it. Please ensure everything added to the `requirements.txt` is being used in the application.

File: .travis.yml (hunk 1 above)
Comment (VivianDoreen) [reply .travis.yml]: OK, let me do that

Comment (VivianDoreen) [reply .travis.yml]: Am done removing "- pip install -U drf-yasg"

File: requirements.txt (hunk 4 above)
Comment (VivianDoreen) [reply requirements.txt]: I have made the necessary changes

File: authors/apps/authentication/migrations/0003_user_is_staff.py (hunk 3 above)
Comment (VivianDoreen) [reply authors/apps/authentication/migrations/0003_user_is_staff.py]: I have made the necessary changes

File: authors/apps/authentication/migrations/0002_remove_user_is_staff.py (hunk 2 above)
Comment (VivianDoreen) [reply authors/apps/authentication/migrations/0002_remove_user_is_staff.py]: I have made the necessary changes

**Review by Opio-Emmanuel-Omona — APPROVED** [review]
LGTM

**Comment by fahadmak** [comment]
LGTM

**Review by fahadmak — APPROVED** [review]
LGTM

**Review by archbishop12 — APPROVED** [review]
@VivianDoreen LGTM I am merging this.

**Pull request merged**